eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+Param(
+  [Parameter(Mandatory=$true)]
+  [string] $azdoOrg, 
+
+  [Parameter(Mandatory=$true)]
+  [string] $azdoProject,
+
+  [Parameter(Mandatory=$true)]
+  [int] $buildDefinitionId,
+
+  [Parameter(Mandatory=$true)]
+  [string] $azdoToken,
+
+  [Parameter(Mandatory=$true)]
+  [string] $githubUser,
+  
+  [Parameter(Mandatory=$true)]
+  [string] $githubPAT,
+
+  [Parameter(Mandatory=$true)]
+  [string] $githubOrg, 
+
+  [Parameter(Mandatory=$true)]
+  [string] $targetRepoName,
+
+  [Parameter(Mandatory=$true)]
+  [string] $barToken, 
+
+  [string] $buildParameters = ''
+)
+
+set-strictmode -version 2.0
+$ErrorActionPreference = 'Stop'
+
+. $PSScriptRoot\..\common\tools.ps1
+. $PSScriptRoot\..\common\darc-init.ps1
+
+$arcadeSdkPackageName = 'Microsoft.DotNet.Arcade.Sdk'
+$arcadeSdkVersion = $GlobalJson.'msbuild-sdks'.$arcadeSdkPackageName
+
+# Get a temporary directory for a test root. Use the agent work folder if running under azdo, use the temp path if not.
+$testRootBase = if ($env:AGENT_WORKFOLDER) { $env:AGENT_WORKFOLDER } else { $([System.IO.Path]::GetTempPath()) }
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+$days = 3
```

Comment:
Should we make this an argument to the script? Can still have this default value.

Reply:
Done!